docs/tutorials/mxnet_models/trainer_callbacks.md

```diff
@@ -0,0 +1,239 @@
+You can use predefined GluonTS callbacks like `TrainingHistory`, `ModelAveraging` or `TerminateOnNaN`, or you can implement your own callback.
+
+```python
+from gluonts.dataset.repository.datasets import get_dataset
+
+dataset = "m4_hourly"
+dataset = get_dataset(dataset)
+prediction_length = dataset.metadata.prediction_length
+freq = dataset.metadata.freq
+```
+
+## Using a single callback
+
+To use callbacks, simply pass them as a list when constructing the `Trainer`:
+in the following example, we are using the `TrainingHistory` callback to record loss values measured during training.
+
+```python
+from gluonts.model.simple_feedforward import SimpleFeedForwardEstimator
+from gluonts.mx import Trainer
+from gluonts.mx.trainer.callback import TrainingHistory
+
+# defining a callback, which will log the training loss for each epoch
+history = TrainingHistory()
+
+trainer = Trainer(epochs=3, callbacks=[history])
+estimator = SimpleFeedForwardEstimator(prediction_length=prediction_length, freq=freq, trainer=trainer)
+
+predictor = estimator.train(dataset.train, num_workers=None)
+```
+
+Print the training loss over the epochs:
+
+```python
+print(history.loss_history)
+```
+
+## Using multiple callbacks
+
+To continue the training from a given predictor you can use the `WarmStart` callback.
+When you want to use more than one callback, just provide a list with multiple callback objects:
+
+```python
+from gluonts.mx.trainer.callback import WarmStart
+
+warm_start = WarmStart(predictor=predictor)
+
+trainer=Trainer(epochs=3, callbacks=[history, warm_start])
+
+estimator = SimpleFeedForwardEstimator(prediction_length=prediction_length, freq=freq, trainer=trainer)
+
+predictor = estimator.train(dataset.train, num_workers=None)
+```
+
+```python
+print(history.loss_history) # The training loss history of all 3+3 epochs we trained the model for
+```
+
+## Default callbacks
+
+In addition to the callbacks you specify, the `Trainer` class uses the two default callbacks `ModelAveraging` and `LearningRateReduction`.
+You can turn them off by setting add_default_callbacks=False when initializing the Trainer.
+
+```python
+trainer=Trainer(epochs=20, callbacks=[history]) # use the TrainingHistory Callback and the default callbacks.
+trainer=Trainer(epochs=20, callbacks=[history], add_default_callbacks=False) # use only the TrainingHistory Callback
+trainer=Trainer(epochs=20, add_default_callbacks=False) # use no callback at all
+```
+
+## Custom callbacks
+
+To implement your own callback you can write a class which inherits from `gluonts.mx.trainer.Callback`, and overwrite one or more of the hooks.
+Have a look at the abstract `Callback` class, the hooks take different arguments which you can use. 
+Hook methods with boolean return value stop the training if False is returned.
+
+Here is an example implementation of a Metric value based early stopping custom callback implementation.
```

Comment:
"Here is an example for a custom callback implementation which terminates training early based on the value of some Metric (such as the RMSE)"?